### ISSUE TITLE
`better-regex` should not be reported when `.toString()` or `String()` is called because it means different

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1878]
```js
rule.test?.toString() === /\.[cm]?[tj]sx?$/.toString()
```

**Comment by fisker** [comment]
I still think we should report that, but I'm going to remove the auto-fix for this case.
